sdk/core/azure-core/CMakeLists.txt

```diff
@@ -42,14 +42,20 @@ if(BUILD_TRANSPORT_CURL)
     src/http/curl/curl_connection_pool_private.hpp
     src/http/curl/curl_connection_private.hpp
     src/http/curl/curl_session_private.hpp
-  )
+    src/http/curl/curl_websockets.cpp
+   )
   SET(CURL_TRANSPORT_ADAPTER_INC
     inc/azure/core/http/curl_transport.hpp
+    inc/azure/core/http/websockets/curl_websockets_transport.hpp
   )
 endif()
 if(BUILD_TRANSPORT_WINHTTP)
-  SET(WIN_TRANSPORT_ADAPTER_SRC src/http/winhttp/win_http_transport.cpp)
-  SET(WIN_TRANSPORT_ADAPTER_INC inc/azure/core/http/win_http_transport.hpp)
+  SET(WIN_TRANSPORT_ADAPTER_SRC 
+    src/http/winhttp/win_http_transport.cpp
```

Comment:
Do we expect that implementing an http transport adapter will now also require providing a websocket implementation?

Reply:
Not really, only if they want websocket support on their transport adapter.

There are still some rough edges in the websocket codebase in this area (specifically: I don't have the hook in place for customer authored websocket transports), but there should be no reason for a customer with a private HTTP transport who doesn't use WebSockets to have to implement WebSockets (that's why the WebSocketXxxTransport derives from the  HttpTransport class).